src/ToolBox/superpmi/superpmi-shared/spmiutil.cpp

```diff
@@ -107,3 +117,110 @@ LPSTR GetCommandLineA()
     return pCmdLine;
 }
 #endif // FEATURE_PAL
+
+void LoadRealJitLib(HMODULE& jitLib, WCHAR* jitLibPath)
+{
+    // Load Library
+    if (jitLib == NULL)
+    {
+        if (jitLibPath == nullptr)
+        {
+            LogError("LoadRealJitLib - No real jit path");
+            return;
+        }
+        jitLib = ::LoadLibraryW(jitLibPath);
+        if (jitLib == NULL)
+        {
+            LogError("LoadRealJitLib - LoadLibrary failed to load '%ws' (0x%08x)", jitLibPath, ::GetLastError());
+            return;
+        }
+    }
+}
+
+void cleanupExecutableName(WCHAR* executableName)
+{
+    WCHAR* quote = nullptr;
+
+    // If there are any quotes in the file name convert them to spaces.
+    while ((quote = wcsstr(executableName, W("\""))) != nullptr)
+    {
+        *quote = W(' ');
+    }
+
+    // Remove any illegal or annoying characters from the file name by converting them to underscores.
+    while ((quote = wcspbrk(executableName, W("=<>:\"/\\|?! *.,"))) != nullptr)
+    {
+        *quote = W('_');
+    }
+}
+
+void generateRandomSuffix(WCHAR* buffer, size_t length)
+{
+    unsigned randNumber = 0;
+#ifdef FEATURE_PAL
```

Comment:
This code is different from the original code:
```
#ifdef FEATURE_PAL
    unsigned __int64 randNumber       = 0;
    const size_t     RandNumberLength = sizeof(randNumber) * 2 + 1; // 16 hex digits + null
    WCHAR            RandNumberString[RandNumberLength];
    PAL_Random(&randNumber, sizeof(randNumber));
    swprintf_s(RandNumberString, RandNumberLength, W("%016llX"), randNumber);
#else  // !FEATURE_PAL
    unsigned int randNumber       = 0;
    const size_t RandNumberLength = sizeof(randNumber) * 2 + 1; // 8 hex digits + null
    WCHAR        RandNumberString[RandNumberLength];
    rand_s(&randNumber);
    swprintf_s(RandNumberString, RandNumberLength, W("%08X"), randNumber);
#endif // !FEATURE_PAL
```

In particular, there's only 8 characters of random for the PAL case instead of 16. Is that ok? does the PAL random function work fine in this case?

Reply:
In methodcallsummarizer.cpp the code used 8 characters. 
According to `coreclr\src\pal\src\misc\miscpalapi.cpp` it should work.